soc/nxp/imxrt/Kconfig

```diff
@@ -0,0 +1,163 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_FAMILY_IMXRT
+	select HAS_SEGGER_RTT if ZEPHYR_SEGGER_MODULE
+	select HAS_PM
+
+if SOC_FAMILY_IMXRT
+
+# Source series Kconfig files first, so SOCs
+# can override the defaults given here
+rsource "*/Kconfig"
+
+# Used for default value in FLASH_MCUX_FLEXSPI_XIP
+DT_CHOSEN_Z_FLASH := zephyr,flash
+DT_COMPAT_FLEXSPI := nxp,imx-flexspi
+# Macros to shorten Kconfig definitions
+DT_CHOSEN_FLASH_NODE := $(dt_chosen_path,$(DT_CHOSEN_Z_FLASH))
+DT_CHOSEN_FLASH_PARENT := $(dt_node_parent,$(DT_CHOSEN_FLASH_NODE))
+
+config FLASH_MCUX_FLEXSPI_XIP
+	bool "MCUX FlexSPI flash access with xip"
+	default $(dt_node_has_compat,$(DT_CHOSEN_FLASH_PARENT),$(DT_COMPAT_FLEXSPI))
+	select XIP
+	help
+	  Allows for the soc to safely initialize the clocks for the
+	  FlexSpi when planning to execute code in FlexSpi Memory.
+
+# Note- When SECOND_CORE_MCUX is set, the dependencies for this Kconfig
+# should be set elsewhere, since the determination of which SOC core
+# requires the boot header is SOC specific.
+config NXP_IMXRT_BOOT_HEADER
+	bool "Boot header"
+	depends on !(BOOTLOADER_MCUBOOT || SECOND_CORE_MCUX)
```

Comment:
This should be `default y`- this will mirror the previous behavior in HVMv1 where boards had this symbol implied by the `CODE_LOCATION` kconfig selection- and will make sure that the RT boards have the required ROM boot header present.